app/vmui/packages/vmui/src/utils/uplot/series.ts

```diff
@@ -1,45 +1,118 @@
-import { MetricResult } from "../../api/types";
+import { MetricBase, MetricResult } from "../../api/types";
 import uPlot, { Series as uPlotSeries } from "uplot";
 import { getNameForMetric, promValueToNumber } from "../metric";
-import { HideSeriesArgs, BarSeriesItem, Disp, Fill, LegendItemType, Stroke, SeriesItem } from "../../types";
-import { baseContrastColors, getColorFromString } from "../color";
-import { getMedianFromArray, getMaxFromArray, getMinFromArray, getLastFromArray } from "../math";
+import { BarSeriesItem, Disp, Fill, ForecastType, HideSeriesArgs, LegendItemType, SeriesItem, Stroke } from "../../types";
+import { anomalyColors, baseContrastColors, getColorFromString } from "../color";
+import { getLastFromArray, getMaxFromArray, getMedianFromArray, getMinFromArray } from "../math";
 import { formatPrettyNumber } from "./helpers";
 
-export const getSeriesItemContext = (data: MetricResult[], hideSeries: string[], alias: string[]) => {
+// Helper function to extract freeFormFields values as a comma-separated string
+export const extractFields = (metric: MetricBase["metric"]): string => {
+  const excludeMetrics = ["__name__", "for"];
+  return Object.entries(metric)
+    .filter(([key]) => !excludeMetrics.includes(key))
+    .map(([key, value]) => `${key}: ${value}`).join(",");
+};
+
+const isForecast = (metric: MetricBase["metric"]) => {
+  const metricName = metric?.__name__ || "";
+  const forecastRegex = new RegExp(`(${Object.values(ForecastType).join("|")})$`);
+  const match = metricName.match(forecastRegex);
+  const value = match && match[0] as ForecastType;
```

Comment:
this match can be empty slice

Reply:
That's okay, here it's important to identify the metrics that relate to vmanomaly.